tools/gatk_applyrecalibration.cwl

```diff
@@ -7,48 +7,52 @@ requirements:
   - class: ShellCommandRequirement
   - class: InlineJavascriptRequirement
   - class: ResourceRequirement
-    ramMin: 7000
-    coresMin: 2
+    ramMin: $(inputs.ram * 1000)
+    coresMin: $(inputs.cpu)
 baseCommand: []
 arguments:
   - position: 0
     shellQuote: false
     valueFrom: >-
-      /gatk --java-options "-Xmx5g -Xms5g"
+      /gatk --java-options "-Xms$(Math.floor((inputs.ram - 1)*1000/1.074-1))m -Xmx$(Math.floor(inputs.ram*1000/1.074-1))m"
       ApplyVQSR
       -O tmp.indel.recalibrated.vcf
       -V $(inputs.input_vcf.path)
       --recal-file $(inputs.indels_recalibration.path)
       --tranches-file $(inputs.indels_tranches.path)
-      -ts-filter-level 99.7
+      -ts-filter-level $(inputs.indel_ts_filter_level)
       --create-output-bam-index true
       -mode INDEL
 
-      /gatk --java-options "-Xmx5g -Xms5g"
+      /gatk --java-options "-Xms$(Math.floor((inputs.ram - 1)*1000/1.074-1))m -Xmx$(Math.floor(inputs.ram*1000/1.074-1))m"
       ApplyVQSR
       -O scatter.filtered.vcf.gz
       -V tmp.indel.recalibrated.vcf
       --recal-file $(inputs.snps_recalibration.path)
       --tranches-file $(inputs.snps_tranches.path)
-      -ts-filter-level 99.7
+      -ts-filter-level $(inputs.snp_ts_filter_level)
       --create-output-bam-index true
       -mode SNP
 inputs:
   input_vcf:
     type: File
-    secondaryFiles: [.tbi]
+    secondaryFiles: [{pattern: '.tbi', required: true}]
   indels_recalibration:
     type: File
-    secondaryFiles: [.idx]
+    secondaryFiles: [{pattern: '.idx', required: true}]
   indels_tranches: File
   snps_recalibration:
     type: File
-    secondaryFiles: [.idx]
+    secondaryFiles: [{pattern: '.idx', required: true}]
   snps_tranches: File
+  snp_ts_filter_level: { type: 'float', doc: "The truth sensitivity level at which to start filtering SNP data" }
+  indel_ts_filter_level: { type: 'float', doc: "The truth sensitivity level at which to start filtering INDEL data" }
+  cpu: { type: 'int?', default: 1, doc: "CPUs to allocate to this task." }
```

Comment:
Why default 1 when old default was 2?

Reply:
It matched it to the current GATK pipeline. They seem to think it doesn't need 2 CPUs, so I freed up the core.